tests/RestApiTestCase.php

```diff
@@ -0,0 +1,195 @@
+<?php
+namespace Give\Tests;
+
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+use WP_Roles;
+use WP_UnitTest_Factory;
+
+/**
+ * Give Unit API Test Case
+ *
+ * Provides Give-specific setup/tear down/assert methods
+ * and helper functions to be used in API tests.
+ *
+ * @unreleased
+ */
+class RestApiTestCase extends TestCase
+{
+    /**
+     * Test REST Server
+     *
+     * @unreleased
+     *
+     * @var WP_REST_Server
+     */
+    protected $server;
+
+    /**
+     * Test user accounts for API authentication
+     *
+     * @unreleased
+     *
+     * @var int[] $users
+     */
+    protected static $users = [
+        'anonymous' => null,
+        'administrator' => null,
+        'editor' => null,
+        'author' => null,
+        'contributor' => null,
+        'subscriber' => null,
+    ];
+
+    /**
+     * Create users for API authentication
+     *
+     * @unreleased
+     *
+     * @return void
+     */
+    public static function wpSetUpBeforeClass(WP_UnitTest_Factory $factory)
+    {
+        self::$users = [
+            'anonymous' => 0,
```

Comment:
What does this user role do?

Reply:
It does nothing. It's such as not authenticated. We force to set current user to 0 to be sure it's not authenticated while dispatching the request.